node/node.go

```diff
@@ -825,6 +825,13 @@ func (node *AlgorandFullNode) loadParticipationKeys() error {
 				return fmt.Errorf("AlgorandFullNode.loadParticipationKeys: cannot load account at %v: %v", info.Name(), err)
 			}
 		} else {
+			// Tell the ParticipationRegistry about the Participation (dupes don't matter)
+			// TODO: Put the registry in the AccountManager?
```

Comment:
```golang
// TODO: Put the registry in the AccountManager?
```

exactly - this entire function is going to be called only upon startup, and therefore it's much better to be moved into the AccountManager.

Reply:
I'll try doing this again. Your suggestion in my last PR to move `registry.Register` call from `OnNewBlock` -> `VotingKeys` should make this a lot cleaner.